docs/CaliptraHardwareSpecification.md

```diff
@@ -2521,6 +2522,116 @@ Data vault is a set of generic scratch pad registers with specific lock function
 * 4B scratchpad registers that are lockable but cleared on warm reset (10 registers)
 * 4B scratchpad registers that are cleared on warm reset (8 registers)
 
+
+## OCP LOCK Hardware Architecture
+
+### 1.1 Overview
+Changes are made to support OCP LOCK flows defined for SSD applications. Specification is published here:   
+[OCP L.O.C.K. Specification](https://chipsalliance.github.io/Caliptra/ocp-lock/specification/HEAD)
+
+### 3 HW config register bit and macro
+**CALIPTRA_OCP_LOCK_EN** constant-value input strap, with a bit in **CPTRA_HW_CONFIG** register:
+- Enables HEK DOE
+- Enables CALIPTRA_IN_OCP_LOCK to be set
+**CLP_DV_REG_NONSTICKYGENERICSCRATCHREG_0**: The least significant bit (LSB) of this register indicates whether the HEK seed is in permanent mode, meaning all HEK seeds have been used and no valid HEK seed remains.
+
+**NOTE:** CALIPTRA_OCP_LOCK_EN is strap pin and needs to be driven  with a constant value by the integrator. 
+
+### 4 HW “mode” register bit
+**CALIPTRA_OCP_LOCK_PROGRESS_REG**
+- KeyVault slots 0-15: Standard use-cases
+- KeyVault slots 16-23: Standard use-cases
+- Write-1-to-lock
+- ROM must set this after performing LOCK derivations (HEK, EPK, etc)
+- CALITPRA_OCP_LOCK_EN + CALIPTRA_MODE_SUBSYSTEM macro is the write-enable for this bit
+- Prevents interactions between slots (STANDARD slots) and (LOCK slots)
+- Enables Key Release operation
+
+### 5.3 Add AES Write Path
+Decrypted Keys (or “unwrapped”/ “ready” keys, e.g. MPK) may be written to KV slots.  
+**MEK** must be written to KV slot 23. Enforced in HW by detecting AES-ECB decrypt operation with KvSlots16 (MDK) as key.
+
+### 5.7 Add KeyVault access rules and filtering logic
+- ROM is responsible for choosing dest slot for HEK seed
+- INSTEAD, Slot 23 PREVENTED as destination for all DOE operations
+  - Slot 23 (place for MEK) restricted so ONLY AES can write to it (when OCP LOCK in progress)
+  - Slot 22 also used to store full derived HEK, which overwrites HEK seed (ROM requirement)
+  - Locked for writes until warm reset (ROM requirement, must lock after each reset)
+  - Locked for copying until warm reset (ROM requirement, must lock after each reset)
+- There must not be a way from KV0-16 to KV10-15 or from KV10-15 to KV9-0 if OCP parameter and OCP_LOCK_PROGRESS_REG, which is set only by ROM, are high.
```

Comment:
KV numbers wrong. Also register is slightly wrong name. You can probably exclude "OCP parameter" since OCP_LOCK_IN_PROGRESS set means OCP parameter is set.